release-tools/check-crate-version/src/main.rs

```diff
@@ -0,0 +1,177 @@
+        })
+        .collect();
+    let mut crate_status: HashMap<String, CrateStatus> =
+        HashMap::with_capacity(status_checks.len());
+
+    for status_check in status_checks {
+        let (crate_name, status) = status_check;
+        let status = status.await.unwrap();
+        crate_status.insert(crate_name, status);
+    }
+    for (crate_name, status) in crate_status.into_iter() {
+        match status {
+            CrateStatus::VersionBumped(manifest_diff) => {
+                println!("🟢 {crate_name:-padding$} Version number has been updated");
+                diff_crate_src(&crate_name, verbose);
+                if verbose {
+                    println!("{manifest_diff}");
+                }
+            }
+            CrateStatus::Unchanged => println!("🟢 {crate_name:-padding$} Code does not differ from crates.io"),
+            CrateStatus::CodeChanged(manifest_diff) => {
+                println!("⛔ {crate_name:-padding$} Code changed but version number did not:");
+                diff_crate_src(&crate_name, true);
+                if let Some(manifest_diff) = manifest_diff {
+                    println!("{manifest_diff}");
+                }
+            },
+            CrateStatus::ManifestChanged(manifest_diff) => {
+                println!("⛔ {crate_name:-padding$} Manifest (Cargo.toml) changed but version number did not:");
+                print!("{manifest_diff}");
+            },
+            CrateStatus::NotPublished => println!("🟡 {crate_name:-padding$} Crate not found in crates.io (Possible cause: Not published yet?)"),
+        }
+    }
+}
+
+async fn check_crate_status(crate_name: String) -> CrateStatus {
+    if !check_crate_published(&crate_name).await {
+        return CrateStatus::NotPublished;
+    }
+    download_crate(&crate_name, "./crates_io").await;
+
+    let manifests = Manifests::read(&crate_name);
+    let manifest_diff = manifests.diff();
+
+    match (
+        diff_crate_src(&crate_name, false),
+        manifest_diff.is_some(),
+        manifests.diff_versions(),
+    ) {
+        (_, _, true) => CrateStatus::VersionBumped(manifest_diff.unwrap()),
+        (false, false, _) => CrateStatus::Unchanged,
+        (true, _, false) => CrateStatus::CodeChanged(manifest_diff),
+        (false, true, false) => CrateStatus::ManifestChanged(manifest_diff.unwrap()),
+    }
+}
+
+fn read_publish_list() -> Vec<String> {
+    #[derive(Deserialize)]
+    struct PublishList {
+        publish_list: Vec<String>,
+    }
+
+    let list_toml = fs::read_to_string(PUBLISH_LIST_PATH).unwrap();
+    let list: PublishList = toml::from_str(&list_toml).unwrap();
+    list.publish_list
+}
+
+/// Checks crates.io to see if the crate has been published. Returns `true` if it has.
+async fn check_crate_published(crate_name: &str) -> bool {
+    // We have to identify ourselves, or crates.io will ignore our request
+    // TODO: Add an email to the user_agent filed - this will make us less likely to have our
```

Comment:
you can use `team@infinyon.com` for crates.io's crawler policy

Reply:
Implemented